src/corehost/cli/deps_format.cpp

```diff
@@ -173,41 +173,51 @@ bool deps_json_t::perform_rid_fallback(rid_specific_assets_t* portable_assets, c
     
     for (auto& package : portable_assets->libs)
     {
-        pal::string_t matched_rid = package.second.rid_assets.count(host_rid) ? host_rid : _X("");
-        if (matched_rid.empty())
+        for (size_t asset_type_index = 0; asset_type_index < deps_entry_t::asset_types::count; asset_type_index++)
         {
-            if (rid_fallback_graph.count(host_rid) == 0)
+            auto& rid_assets = package.second[asset_type_index].rid_assets;
+            pal::string_t matched_rid = rid_assets.count(host_rid) ? host_rid : _X("");
+            if (matched_rid.empty())
             {
-                trace::warning(_X("The targeted framework does not support the runtime '%s'. Some native libraries from [%s] may fail to load on this platform."), host_rid.c_str(), package.first.c_str());
-            }
-            else
-            {
-                const auto& fallback_rids = rid_fallback_graph.find(host_rid)->second;
-                auto iter = std::find_if(fallback_rids.begin(), fallback_rids.end(), [&package](const pal::string_t& rid) {
-                    return package.second.rid_assets.count(rid);
-                });
-                if (iter != fallback_rids.end())
+                if (rid_fallback_graph.count(host_rid) == 0)
                 {
-                    matched_rid = *iter;
+                    trace::warning(_X("The targeted framework does not support the runtime '%s'. Some native libraries from [%s] may fail to load on this platform."), host_rid.c_str(), package.first.c_str());
+                }
+                else
+                {
+                    const auto& fallback_rids = rid_fallback_graph.find(host_rid)->second;
+                    auto iter = std::find_if(fallback_rids.begin(), fallback_rids.end(), [&rid_assets](const pal::string_t& rid) {
+                        return rid_assets.count(rid);
+                        });
+                    if (iter != fallback_rids.end())
+                    {
+                        matched_rid = *iter;
+                    }
                 }
             }
-        }
-
-        if (matched_rid.empty())
-        {
-            package.second.rid_assets.clear();
-        }
 
-        for (auto iter = package.second.rid_assets.begin(); iter != package.second.rid_assets.end(); /* */)
-        {
-            if (iter->first != matched_rid)
+            if (matched_rid.empty())
             {
-                trace::verbose(_X("Chose %s, so removing rid (%s) specific assets for package %s"), matched_rid.c_str(), iter->first.c_str(), package.first.c_str());
-                iter = package.second.rid_assets.erase(iter);
+                rid_assets.clear();
             }
-            else
+
+            for (auto iter = rid_assets.begin(); iter != rid_assets.end(); /* */)
```

Comment:
Can't you use std::remove_if() here?  You're already using std::find_if() in the same function.

Reply:
I guess I don't know which part you're referring to. The for loop around rid_assets is removing keys from a map. `remove_if` doesn't work on maps. In fact the implemented for loop looks exactly like the best suggested solutions how to do such a thing on SO.